pkg/boot/multiboot/module.go

```diff
@@ -102,22 +102,29 @@ func alignUp(buf *bytes.Buffer) error {
 	return err
 }
 
-func (m *module) loadModule(buf *bytes.Buffer, name string) error {
+func (m *module) loadModule(buf *bytes.Buffer, r io.ReaderAt, name string) error {
 	log.Printf("Adding module %v", name)
 
-	b, err := readFile(name)
-	if err != nil {
-		return err
-	}
-
 	// place start of each module to a beginning of a page.
 	if err := alignUp(buf); err != nil {
 		return err
 	}
 
 	m.Start = uint32(buf.Len())
-	if _, err := buf.Write(b); err != nil {
-		return err
+
+	if wtr, ok := r.(io.WriterTo); ok {
```

Comment:
Just use io.Copy(buf, r)? io.Copy does all this under the hood afaik.

Reply:
Thanks, I often miss some basic functions from the standard library (probably a bias from my C background)